### PR TITLE
[ie/RTP] Support more subpages

### PR DESCRIPTION
**Pull request by Demon000**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

On the zigzag subdomain, the file property for the RTPPlayer constructor object is inlined, as opposed to being declared before the RTPPlayer instantiation. Add support for this.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/rtp.py [hunk 1]
```diff
@@ -49,18 +59,21 @@ def _real_extract(self, url):
...
-        f = self._parse_json(
+        f = None if f is None else self._parse_json(
             f, video_id,
             lambda data: self.__unobfuscate(data, video_id=video_id))
         config = self._parse_json(
             config, video_id,
             lambda data: self.__unobfuscate(data, video_id=video_id))
 
+        if f is None:
+            f = config.get('file')
```
Comment (bashonly) [inline yt_dlp/extractor/rtp.py:75]: if we flip-flop the order of JSON parsing we can simplify this
```suggestion
        config = self._parse_json(
            config, video_id,
            lambda data: self.__unobfuscate(data, video_id=video_id))
        f = config.get('file') if not f else self._parse_json(
            f, video_id,
            lambda data: self.__unobfuscate(data, video_id=video_id))
```

File: yt_dlp/extractor/rtp.py [hunk 2]
```diff
@@ -49,18 +59,21 @@ def _real_extract(self, url):
 
         f, config = self._search_regex(
             r'''(?sx)
-                var\s+f\s*=\s*(?P<f>".*?"|{[^;]+?});\s*
+                (var\s+f\s*=\s*(?P<f>".*?"|{[^;]+?});\s*)?
```
Comment (bashonly) [inline yt_dlp/extractor/rtp.py:62]: non-capturing group
```suggestion
                (?:var\s+f\s*=\s*(?P<f>".*?"|{[^;]+?});\s*)?
```

File: yt_dlp/extractor/rtp.py [hunk 3]
```diff
@@ -8,7 +8,7 @@
 
 
 class RTPIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?rtp\.pt/play/p(?P<program_id>[0-9]+)/(?P<id>[^/?#]+)/?'
+    _VALID_URL = r'https?://(?:www\.)?rtp\.pt/play(/zigzag)?/p(?P<program_id>[0-9]+)/(?P<id>[^/?#]+)/?'
```
Comment (bashonly) [inline yt_dlp/extractor/rtp.py:11]:
```suggestion
    _VALID_URL = r'https?://(?:www\.)?rtp\.pt/play/(?:zigzag/)?p(?P<program_id>[0-9]+)/(?P<id>[^/?#]+)/?'
```

**Review by bashonly — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/rtp.py [hunk 4]
```diff
@@ -8,7 +8,7 @@
 
 
 class RTPIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?rtp\.pt/play/p(?P<program_id>[0-9]+)/(?P<id>[^/?#]+)/?'
+    _VALID_URL = r'https?://(?:www\.)?rtp\.pt/play/(?:zigzag/)?p(?P<program_id>[0-9]+)/(?P<id>[^/?#]+)/?'
```
Comment (seproDev) [inline yt_dlp/extractor/rtp.py:11]: Can you add `estudoemcasa` and `palco` as well?
```suggestion
    _VALID_URL = r'https?://(?:www\.)?rtp\.pt/play/(?:(?:estudoemcasa|palco|zigzag)/)?p(?P<program_id>[0-9]+)/(?P<id>[^/?#]+)'
```
Please also add appropriate tests. match_only is fine.
 - https://www.rtp.pt/play/estudoemcasa/p7776/portugues-1-ano
 - https://www.rtp.pt/play/palco/p13785/l7nnon

**Review by seproDev — CHANGES_REQUESTED** [review]
Please don't force push in the future

**Review by seproDev — APPROVED** [review]

**Review by bashonly — APPROVED** [review]

File: yt_dlp/extractor/rtp.py [hunk 5]
```diff
@@ -49,17 +65,17 @@ def _real_extract(self, url):
...
             ''', webpage,
             'player config', group=('f', 'config'))
 
-        f = self._parse_json(
-            f, video_id,
-            lambda data: self.__unobfuscate(data, video_id=video_id))
         config = self._parse_json(
             config, video_id,
             lambda data: self.__unobfuscate(data, video_id=video_id))
+        f = config.get('file') if not f else self._parse_json(
```
Comment (bashonly) [inline yt_dlp/extractor/rtp.py:76]:
```suggestion
        f = config['file'] if not f else self._parse_json(
```

**Pull request merged**